pkg/pstore/api.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pstore // import "storj.io/storj/pkg/pstore"
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+type argError struct {
+	arg string
+	msg string
+}
+
+func (e *argError) Error() string {
+	return fmt.Sprintf("HashError (%s): %s", string(e.arg), e.msg)
+}
+
+type fsError struct {
```

Comment:
same comment about github.com/zeebo/errs, which will not only do most of this for you but also give you stack traces if you want them